File: Arrays and Hashing/242-Valid-Anagram/242.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if(s.Length != t.Length){
            return false;
        }
        Dictionary<char, int> s_map = new Dictionary<char,int>();
        foreach(var letter in s){
            if(s_map.ContainsKey(letter)){
                s_map[letter]++;
            }
            else{
                s_map[letter] = 1;
            }
        }
        foreach(var letter in t){
            if(s_map.ContainsKey(letter)){
                if(s_map[letter] == 0){
                    return false;
                }
                else{
                    s_map[letter]--;
                }
            }
            else{
                return false;
            }
        }
        return true;
    }
}

File: Arrays and Hashing/242-Valid-Anagram/242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(t) != len(s):
            return False
        s_map = {}
        for letter in s:
            if letter in s_map:
                s_map[letter] += 1
            else:
                s_map[letter] = 1
        for letter in t:
            if letter in s_map:
                if s_map[letter] == 0:
                    return False
                else:
                    s_map[letter] -= 1
            else:
                return False
        return True